composeApp/src/main/kotlin/io/github/couchtracker/ui/components/TimezonePickerDialog.kt

```diff
@@ -0,0 +1,129 @@
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.darkColorScheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import io.github.couchtracker.R
+import io.github.couchtracker.utils.str
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.offsetAt
+
+@Composable
+fun TimezonePickerDialog(
+    timezone: TimeZone?,
+    onTimezoneSelected: (TimeZone) -> Unit,
+    close: () -> Unit,
+) {
+    val instant = remember { Clock.System.now() }
+    val timezones = remember { getAllTz(instant) }
+    var selected by remember { mutableStateOf(timezone) }
+    AlertDialog(
+        onDismissRequest = close,
+        icon = { Icon(Icons.Default.TravelExplore, contentDescription = null) },
+        title = { Text(R.string.select_time_zone.str()) },
+        text = { TimeZoneList(timezones, selected, instant) { selected = it } },
+        confirmButton = {
+            TextButton(
+                enabled = selected != null,
+                onClick = {
+                    onTimezoneSelected(checkNotNull(selected))
+                    close()
+                },
+                content = { Text(R.string.select_action.str()) },
+            )
+        },
+        dismissButton = {
+            TextButton(onClick = close) { Text(R.string.cancel_action.str()) }
+        },
+    )
+}
+
+@Composable
+@OptIn(ExperimentalFoundationApi::class)
+private fun TimeZoneList(
+    timezones: List<List<TimeZone>>,
+    selected: TimeZone?,
+    instant: Instant,
+    select: (TimeZone) -> Unit,
+) {
+    // Resetting the theme, as colors are a bit weird otherwise
+    MaterialTheme(colorScheme = darkColorScheme()) {
+        val scrollState = rememberLazyListState()
+        BoxWithConstraints {
+            val h = this.constraints.maxHeight
+            LaunchedEffect(Unit) {
+                // Linearizing the elements with their header
+                val linearized = timezones.flatMap { listOf(null) + it }
+                scrollState.scrollToItem(
+                    index = linearized.indexOf(selected).coerceAtLeast(0),
+                    scrollOffset = -h / 2,
+                )
+            }
+            LazyColumn(state = scrollState) {
+                for (zoneGroup in timezones) {
+                    stickyHeader {
+                        Column(Modifier.background(AlertDialogDefaults.containerColor)) {
+                            Text(
+                                zoneGroup.first().offsetAt(instant).toString(),
```

Comment:
Suggestion: if you use https://developer.android.com/reference/android/icu/util/TimeZone you can use `getDisplayName()` to show a friendly name. The ID can still be shown as a subtitle.

I think you should be able to convert freely between Kotlin's `TimeZone`, Java's `ZoneId` and Android's `TimeZone`.